//promises

then 
catch

//const p=new Promise((resolve,reject)=>{
    setTimeout()={
        resolve("success")
    },200;
})
p.then(console.log)

//const p=new promise((resolve,reject)=>{
    setTimeout()=>{
        reject("failure")
    },2000;
})
p.then(console.log).catch(console.log)